### PR TITLE
Improve the gitignore default

### LINKED ISSUE #57: Add OS-specific files to .gitignore
I think that the default .gitignore file DrWatson generates should include the following OS-specific patterns:

.DS_Store
.DS_Store?
._*
.Spotlight-V100
.Trashes
ehthumbs.db
Thumbs.db

Also, ignore .ipynb_checkpoints for Jupyter Notebooks.
See 
https://stackoverflow.com/questions/107701/how-can-i-remove-ds-store-files-from-a-git-repository
https://stackoverflow.com/questions/36306017/should-ipynb-checkpoints-be-stored-in-git

### PR DESCRIPTION
**Pull request by asinghvi17**
Adds a lot of new files to ignore!

Fixes #57

### CODE REVIEW

**Comment by asinghvi17** [comment]
Might be a _little_ big :D

File: src/defaults/gitignore.txt [hunk 1]
```diff
@@ -1,24 +1,426 @@
...
+
+# Build artifacts for creating documentation generated by the Documenter package
+docs/build/
+docs/site/
+
+# File generated by Pkg, the package manager, based on a corresponding
+# `Project.toml`.  It records a fixed state of all packages used by the project.
+# As such, it should not be committed for packages, but should be committed for
+# applications that require a static environment.
+Manifest.toml
```
Comment (Datseris) [inline src/defaults/gitignore.txt]: Manifest.toml is absolutely crucial for reproducibility of scientific results and should always be committed.

File: src/defaults/gitignore.txt [hunk 2]
```diff
@@ -1,24 +1,426 @@
...
+
+################################################################################
+#                               Jupyter Notebook                               #
+################################################################################
+
+.ipynb_checkpoints
+*/.ipynb_checkpoints/*
+
+# Remove previous ipynb_checkpoints
```
Comment (Datseris) [inline src/defaults/gitignore.txt]: What do these comments here mean? Are they necessary?

Comment (asinghvi17) [reply src/defaults/gitignore.txt]: No, not really - it's just a command to remove all `.ipynb_checkpoints` from git history.  It's useful if you're using DrWatson on a previously initialized git repo, but other than that there's not a need for it.

File: src/defaults/gitignore.txt (hunk 1 above)
Comment (asinghvi17) [reply src/defaults/gitignore.txt]: Will change that, thanks!

File: src/defaults/gitignore.txt (hunk 2 above)
Comment (Datseris) [reply src/defaults/gitignore.txt]: I think this comment then doesn't really belong in this `.gitignore` file.

Comment (asinghvi17) [reply src/defaults/gitignore.txt]: Made the change, thanks!

**Comment by Datseris** [comment]
Can you please also increment version in `Project.toml`?

**Pull request merged**